### PR TITLE
Fix the output message in SGEN.

### LINKED ISSUE #23376: Review the output message in SGEN
Need review the output message in SGEN since lots of text just copied from desktop version.

### PR DESCRIPTION
**Pull request by huanwu**
Using word "file" instead of "assembly" since the sgen will generate a cs file not an assembly.

Fix #23376

@shmao @zhenlan @mconnew

### CODE REVIEW

**Comment by zhenlan** [comment]
The fact that sgen has .cs generated is just implementation detail. It is a intermediate file. What customers see (and get) is still an assembly.

File: src/System.Private.Xml/src/Resources/Strings.resx [hunk 1]
```diff
@@ -3346,19 +3346,19 @@
...
-    <value>Cannot generate serialization assembly {0} because it already exists. Use /{1} to force an overwrite of the existing assembly.</value>
+    <value>Cannot generate serialization file {0} because it already exists. Use /{1} to force an overwrite of the existing file.</value>
   </data>
   <data name="ErrDirectoryExists" xml:space="preserve">
-    <value>Cannot generate serialization assembly '{0}' because a directory with the same name already exists.</value>
+    <value>Cannot generate serialization file '{0}' because a directory with the same name already exists.</value>
   </data>
   <data name="ErrDirectoryNotExists" xml:space="preserve">
-    <value>Cannot generate serialization assembly '{0}' because directory {1} doesn't exist.</value>
+    <value>Cannot generate serialization file '{0}' because directory {1} doesn't exist.</value>
```
Comment (danmosemsft) [inline src/System.Private.Xml/src/Resources/Strings.resx]: single quotes around {1}

File: src/System.Private.Xml/src/Resources/Strings.resx [hunk 2]
```diff
@@ -3346,19 +3346,19 @@
     <value>Uppercase-First sorting option is not supported.</value>
   </data>
   <data name="XmlPregenTypeDynamic" xml:space="preserve">
-    <value>Cannot pre-generate serialization assembly for type '{0}'. Pre-generation of serialization assemblies is not supported for dynamic types. Save the assembly and load it from disk to use it with XmlSerialization.</value>
+    <value>Cannot pre-generate serialization file for type '{0}'. Pre-generation of serialization assemblies is not supported for dynamic types. Save the assembly and load it from disk to use it with XmlSerialization.</value>
   </data>
   <data name="XmlPregenOrphanType" xml:space="preserve">
     <value>Cannot pre-generate serializer for multiple assemblies. Type '{0}' does not belong to assembly {1}.</value>
```
Comment (danmosemsft) [inline src/System.Private.Xml/src/Resources/Strings.resx]: single quotes around {1} ?

**Comment by huanwu** [comment]
@zhenlan The tool will firstly generate the .cs. We need output the exact error we got. As for the assembly, if there's any failure on build an assembly e.g. got error in the build task, we still will output that error.

File: src/System.Private.Xml/src/Resources/Strings.resx (hunk 2 above)
Comment (huanwu) [reply src/System.Private.Xml/src/Resources/Strings.resx]: Yes, all follow this format in Strings.resx.

**Comment by huanwu** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by zhenlan** [comment]
> @zhenlan The tool will firstly generate the .cs. We need output the exact error we got.

If we want to be specific, instead of saying "file", why don't we say "code"?

**Pull request merged**